### ISSUE TITLE
Bookmark sync support?

### ISSUE THREAD

**Issue by orschiro** [yguan/bookmark-tagging#1]
Hi there,

Thanks for developing this extremely promising extension!

If I use your extension to create bookmarks, are these recognised as normal Chrome bookmarks and thus synced by Chrome across several machines?

Cheers!

**Comment by yguan** [comment]
No, it doesn't sync the bookmarks across machines. It stores the bookmark in the indexedDB maintained by your browser. Chrome may not support syncing of indexedDB in the distant future because indexedDB's capacity is determined by the available space in your storage drive.

**Comment by orschiro** [comment]
Thanks for clarification.

Bummer to hear that. At least for me, cross-machine support is more important to ensure than tagging.

**Comment by lsimone** [comment]
hello @yguan , the @orschiro concern is mine too: I think synchronization is the most important feature this extension lacks.
there is [another chrome extension](https://chrome.google.com/webstore/detail/bookmark/abagojmcbfaibjonagkfhklflndpcjpd?hl=en) that stores the bookmarks into multiple folders (one for each tag), so that, native Chrome synchronization will bring in all your machines your tagged bookmarks library.

Isn't it an interesting enhancement idea?

Since I really appreciate your work, I would also propose to contribute, if you need help.
have a nice day!

**Comment by yguan** [comment]
Hi @lsimone, the limitation with the current implementation is its use of indexedDB. Your proposal of using bookmark to store tag is a good idea, and I think store all bookmarks in one folder, and store all tags and bookmark IDs in another folders may be the way to go. When the app is loaded, it will load all tags and bookmark IDs and the subsequent operation will be in memory.

I don't have any time to work on this extension for the next 3 months. If you're interested in helping out, please fork it and request merge once it's done, and your name will be mentioned on the github repository and chrome store.

The code that you may need to change should be restricted to bookmark-repository.js, tag-group-repository.js, and bookmark-loader.js.

**Comment by lsimone** [comment]
Hi @yguan , 
did you ever have a look at: https://developer.chrome.com/apps/storage ?
It can be another implementation idea.

**Comment by yguan** [comment]
Yes. The problem with storage is the max capacity is 102,400 bytes.

**Comment by lsimone** [comment]
uhm...isn't that enough for a reasonable bookmarks collection?

**Comment by TylerRick** [comment]
Any update on this?

By the way, I came across https://github.com/akshaykumar90/savory, which looks like it will allow sync support ... but it doesn't appear to be very finished or usable yet.
